geomesa-fs/geomesa-fs-tools/src/main/scala/org/locationtech/geomesa/fs/tools/compact/FileSystemCompactionJob.scala

```diff
@@ -77,7 +80,14 @@ trait FileSystemCompactionJob extends StorageConfiguration with JobWithLibJars {
     configureOutput(storage.metadata.sft, job)
 
     // save the existing files so we can delete them afterwards
-    val existingDataFiles = partitions.map(p => (p, storage.getFilePaths(p.name).toList)).toList
+    // mimic the filtering done in PartitionInputFormat
+    val sizeable = Option(storage).collect { case s: SizeableFileSystemStorage => s }
+    val sizeCheck = sizeable.flatMap(s => s.targetSize(targetFileSize).map(t => (p: Path) => s.fileIsSized(p, t)))
+    val existingDataFiles = partitions.toList.flatMap { p =>
+      val files = storage.getFilePaths(p.name).filterNot(f => sizeCheck.exists(_.apply(f.path)))
+      // TODO get counts right... use m/r counters?
```

Comment:
Is there a ticket we could enter for this?

Reply:
i'm not sure it matters at the moment - we should probably revisit this more holistically as part of GEOMESA-2961, 2962 and 2963 (i.e. if/when we start using the metadata values for query file pruning). I updated 2962 with a note about this.